File: packages/lib-model-js/src/test/TestableEntityResource/TestableEntityResource.models.ts
import { type TestableEmbeddedResourceModel } from '@lib/model/test/TestableEmbeddedResource/TestableEmbeddedResource.models';
import { type TestableRelatedResourceModel } from '@lib/model/test/TestableRelatedResource/TestableRelatedResource.models';
import { type TestableResourceModel } from '@lib/model/test/TestableResource/TestableResource.models';
import { type PartialArrayModel } from '@lib/shared/core/core.models';

export type TestableEntityResourceModel = TestableResourceModel & {
  embedded?: PartialArrayModel<TestableEmbeddedResourceModel>;

  relatedManyToMany?: PartialArrayModel<TestableRelatedResourceModel>;

  relatedOneToMany?: PartialArrayModel<TestableRelatedResourceModel>;
};

File: packages/lib-model-py/src/lib_model/test/testable_entity_resource/testable_entity_resource_models.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, RootModel

from ..testable_embedded_resource import testable_embedded_resource
from ..testable_related_resource import testable_related_resource


class RefModelTestableEntityResourceModel(BaseModel):
    pass
    model_config = ConfigDict(
        extra='allow',
    )


class Model(RootModel[TestableEntityResourceModel]):
    root: TestableEntityResourceModel


class TestableEntityResourceModel(BaseModel):
    model_config = ConfigDict(
        extra='allow',
    )
    embedded: Optional[
        List[testable_embedded_resource.TestableEmbeddedResourceModel]
    ] = None
    relatedManyToMany: Optional[
        List[testable_related_resource.TestableRelatedResourceModel]
    ] = None
    relatedOneToMany: Optional[
        List[testable_related_resource.TestableRelatedResourceModel]
    ] = None
    date: Optional[datetime] = None
    group: str
    index: float
    number: Optional[float] = None
    string: str
    stringArray: Optional[List[str]] = None
    stringOptional: Optional[str] = None
    field_id: str
    created: datetime
    isFixture: Optional[bool] = None
    beforeCreate: None = None


class CollectionModelTestableEntityResourceModel(
    RootModel[List[TestableEntityResourceModel]]
):
    root: List[TestableEntityResourceModel]


Model.model_rebuild()
TestableEntityResourceModel.model_rebuild()
